File: src/main/java/a6n_test_001/solution.py
##!/bin/python3

import math
import os
import random
import re
import sys

from urllib.request import urlopen
import json



#
# Complete the 'getUsernames' function below.
#
# The function is expected to return a STRING_ARRAY.
# The function accepts INTEGER threshold as parameter.
#
# URL for cut and paste
# https://jsonmock.hackerrank.com/api/article_users?page=<pageNumber>
#

def getUsernames(threshold: int):
    # Write your code here
    def get_url(page_number: int):
        return "https://jsonmock.hackerrank.com/api/article_users?page={0}".format(page_number)
    
    class Author(object):
        def __init__(self, input_json):
            self.__dict__ = json.loads(input_json)
        
        id: int
        username: str
        about: str
        submitted: int
        updated_at: str
        submission_count: int
        comment_count: int
        created_at: str

    class Authors(object):
        def __init__(self, input_json):
            self.__dict__ = json.loads(input_json)
        
        page: int
        per_page: int
        total: int
        total_pages: int
        data: [Author]

    
    print(get_url(2))
    print(urlopen(get_url(1)))
    res = urlopen(get_url(1))
    bbb = res.read()
    print(bbb.decode("utf8"))
    
    print(json.loads(bbb))
    result = json.loads(bbb)
    # result2 = json.loads(res.read())
    result2 = Authors(bbb)
    res.close()

    print("11111")
    print(result)
    print(result2.page)
    print(result2.per_page)
    print(result2.total_pages)
    print(result2.data)
    print(result2.data[0])
    # print("2222")
    # print(result2)


if __name__ == '__main__':
    getUsernames(10)

File: src/main/java/a6n_test_001/Solution.java
package a6n_test_001;

import static java.util.stream.Collectors.joining;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
import java.util.function.*;

@SuppressWarnings("PMD")
class Result {

    /*
     * Complete the 'getUsernames' function below.
     *
     * The function is expected to return a STRING_ARRAY.
     * The function accepts INTEGER threshold as parameter.
     *
     * URL for cut and paste
     * https://jsonmock.hackerrank.com/api/article_users?page=<pageNumber>
     */

    public static List<String> getUsernames(int threshold) throws IOException {
        final Function<Integer, String> getUsersByPageNumber =
                (n) ->
                        String.format(
                                "https://jsonmock.hackerrank.com/api/article_users?page=%s", n);

        System.out.println(getUsersByPageNumber.apply(1));

        java.net.HttpURLConnection conn =
                (java.net.HttpURLConnection)
                        new java.net.URL(getUsersByPageNumber.apply(1)).openConnection();
        conn.setRequestMethod("GET");

        conn.setDoOutput(true);

        conn.setRequestProperty("Content-Type", "application/json");
        String contentType = conn.getHeaderField("Content-Type");
        conn.setInstanceFollowRedirects(false);

        // DataOutputStream out = new DataOutputStream(conn.getOutputStream());
        // out.flush();
        // out.close();

        int status = conn.getResponseCode();
        System.out.print("status code: " + status);

        BufferedReader in = new BufferedReader(new InputStreamReader(conn.getInputStream()));
        String inputLine;
        StringBuffer content = new StringBuffer();
        while ((inputLine = in.readLine()) != null) {
            content.append(inputLine);
        }
        in.close();

        System.out.println(content);

        return new ArrayList<>();
    }
}

@SuppressWarnings("PMD")
public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter =
                new BufferedWriter(
                        new FileWriter(
                                "D:\\Projects\\Java\\1.txt")); // System.getenv("OUTPUT_PATH")));

        int threshold = Integer.parseInt(bufferedReader.readLine().trim());

        List<String> result = Result.getUsernames(threshold);

        bufferedWriter.write(result.stream().collect(joining("\n")) + "\n");

        bufferedReader.close();
        bufferedWriter.close();
    }
}
